### ISSUE TITLE
CMS Configurable Default Endpoint

### ISSUE THREAD

**Issue by NightJar** [silverstripe/silverstripe-ckan-registry#38]
If a user gives a resource identifier unqualified (with no domain in the URL) to the `ResourceLocatorField`, it will use the configured default endpoint (read: domain which points to a CKAN instance).

The default endpoint can be configured either as a parameter to the `ResourceLocatorField` (by contructor or setter), or is looked up via yaml config for the same.

Both of these currently rely on developer intervention to be set.

It would be great if a content editor could set the default endpoint for either a page or a site, and have unqualified resource IDs use this value instead for looking up (falling back to the configured default if both are unset).

The order of precedence could be:
- Page defined endpoint
- SiteConfig defined endpoint
- Configuration defined endpoint

Where using the setter on the field should _always_ override this "lookup" functionality.

**Comment by robbieaverill** [comment]
It makes sense - CWP is likely to set a default as data.govt.nz for example

**Comment by robbieaverill** [comment]
I thought about it again and I'm unsure we need to provide both a Page and SiteConfig CMS config option. The developer config option exists already, so I think we could just do one or other TBH. I can see why you might want both but I'm unsure if it's worth the UI bloat

**Comment by ScopeyNZ** [comment]
Yeah we've already worked in YAML config for a developer used but @camfindlay indicated that agencies might like the benefit of CMS admins also being able to specify a default.

**Comment by NightJar** [comment]
To be clear this is an extreme "nice to have" at the moment. It's unlikely that we'll be making this change soon, the issue was opened to remind us/track it :)
